napari/plugins/hook_specifications.py

```diff
@@ -317,3 +322,35 @@ def napari_write_vectors(path: str, data: Any, meta: dict) -> Optional[str]:
         If data is successfully written, return the ``path`` that was written.
         Otherwise, if nothing was done, return ``None``.
     """
+
+
+# -------------------------------------------------------------------------- #
+#                                 GUI Hooks                                  #
+# -------------------------------------------------------------------------- #
+
+
+@napari_hook_specification(historic=True)
+def napari_experimental_provide_function_widget() -> Union[
+    MagicFunctionArg, List[MagicFunctionArg]
+]:
+    """Provide functions and args that can be passed to magicgui.
+
+    Returns
+    -------
+    functions : tuple of callable, dict or list tuple of callable, dict
+        List of 3-tuple, where each tuple has a callable function a dictionary of
+        keyword arguments for magicgui.
```

Comment:
This is out of date, btw. I think we can finalise it now?

Reply:
Yeah, I think we're just waiting to see if we go with `MagicFunctionArg` or not and how much to explain here

Reply:
improved